src/compiler/resolutionCache.ts

```diff
@@ -0,0 +1,316 @@
+/// <reference path="types.ts"/>
+/// <reference path="core.ts"/>
+
+namespace ts {
+    export interface ResolutionCache {
+        setModuleResolutionHost(host: ModuleResolutionHost): void;
+
+        startRecordingFilesWithChangedResolutions(): void;

```

Comment:
Instead of two separate functions, how about `recordFilesWithChangedResolutions(action: () => void): Path[];`?